docs/tagging.md

```diff
@@ -106,10 +113,58 @@ You can use [`tag-text.py`](../api-examples/tag-text.py) to load a sequence tagg
+- Use the mead layers (8 mile) API to define your layers
+  - This will minimize any incompatibility and make it easy to switch frameworks later
+- Use `TransducerTaggerModel` for simple models involving overriding the `init_encode()` method.  If you can do this, avoid overidding `create_layers()`
+
+### Some Notes on the TensorFlow implementation
+
+### Eager vs Declarative
+
+In TensorFlow, we are supporting eager and declarative mode models.  This fundamentally changes how our objects must look, as in declarative mode, we must do a `session.run` identifying the graph outputs, vs eager mode, which can look nearly identical to PyTorch.
+
+If you wanted to write code to only support graph mode, especially prior to Keras integration, it was common to simply define the graph procedurally and expose its graph endpoints to the class for future use in `session.run`.  This is what previous versions of MEAD-Baseline did.  With Keras now becoming the primary and recommended interface to building neural networks in TensorFlow execution occurs in essentially 2 phases (initialization and execution) corresponding to creation of the layer by calling its constructor vs execution of the `call()` operator `layer(x)`.  In declarative mode, however, ultimately we do need something to call `session.run` on, so we typically will save the output as a property on the class that can be handed to that call.  In eager mode, storing an output property graph node makes no sense at all, so we just dont do it.  Also in eager mode with 2.x, we save and load models somewhat differently than in TF 1.x.
+
+Eager mode massively simplifies the experience of TensorFlow users, however it is not completely trivial to support a training loop for both side-by-side.  In an ideal world where everyone uses eager, we could just use a normal for loop over the input dataset to train the model with gradient tape.  Of course, this doesnt work with declarative mode.  We decided that the cleanest way to support both methods was to have a single classifier model, but to have multiple custom training loops that are defined as `fit_func`s.  The default `fit_func` for eager mode is the previously mentioned simple for loop with gradient tape whereas, for declarative model it a dataset based method relying on iterators.  In V1.x of MEAD-Baseline, the default `fit_func` was a `feed_dict` method.  We still support this method of training by providing a built-in `fit_func` with key `feed_dict`.
+
+When `mead-train` runs in declarative mode (which is controlled by the `--prefer_eager <bool>` argument), the training block of the MEAD config can contain a key `fit_func: feed_dict` to switch the training loop implementation to the previous default method (this can also be done at run-time by passing `--fit_func feed_dict` as a command-line argument to `mead-train`.
```

Comment:
Should this override be `--x:train.fit_func feed_dict`? I didn't thing we had on off overrides like this anymore

Reply:
TBH, I forgot this functionality existed at the time I wrote it.  Im ok with reverting this part, although the alternative is pretty verbose

Reply:
I guess my gut feeling is it doesnt do any harm.  People who like the long confusing syntax can still use it, but it looks like overriding the `fit_func` will become a lot more common so I dont feel super bad about having modified the driver.